.circleci/config.yml

```diff
@@ -0,0 +1,207 @@
+
+            - run:
+                  name: dependencies
+                  command: |
+                    sudo apt-get update
+                    sudo apt-get install -y software-properties-common curl \
+                    python-software-properties
+                    sudo add-apt-repository -y ppa:hvr/ghc
+                    sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+                    sudo add-apt-repository -y ppa:avsm/ppa
+                    echo 'export TERM=dumb' >> $BASH_ENV
+                    echo 'export LANG=en_US.UTF-8' >> $BASH_ENV
+                    echo 'export LANGUAGE=en_US:en' >> $BASH_ENV
+                    echo 'export LC_ALL=en_US.UTF-8' >> $BASH_ENV
+                    curl -sL https://deb.nodesource.com/setup_8.x |
+                    sudo -E bash  -
+                    curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg |
+                    sudo apt-key add -
+                    echo "deb https://dl.yarnpkg.com/debian/ stable main" |
+                    sudo tee /etc/apt/sources.list.d/yarn.list
+                    sudo apt-get update
+                    sudo apt-get install -y python3.4-venv git bzip2 \
+                    cabal-install-1.24 clang-3.4 libreadline-dev libblas-dev \
+                    gfortran ghc happy indent liblapack-dev language-pack-en \
+                    libperl-critic-perl libxml2-utils menhir php-codesniffer \
+                    build-essential texinfo libbz2-dev libcurl4-openssl-dev \
+                    libexpat-dev sqlite3 zlib1g-dev git-core ruby libssl-dev \
+                    libsqlite3-dev libxslt1-dev libffi-dev yarn python3-dev \
+                    libgdbm-dev libncurses5-dev automake libtool bison \
+                    libncursesw5-dev libc6-dev tk-dev libncurses-dev \
+                    ca-certificates
+
+            - run:
+                  name: virtual-env
+                  command: |
+                    if [ ! -d "$HOME/.venv" ]; then
+                    python3 -m venv ~/.venv
+                    fi
+                    echo 'source $HOME/.venv/bin/activate' >> $BASH_ENV
+
+            - run:
+                  name: pre
+                  command: |
+                    sudo rm -rf /var/cache/apt/archives
+                    sudo ln -s $HOME/.apt-cache /var/cache/apt/archives
+                    mkdir -p $HOME/.apt-cache/partial
+                    mkdir -p $HOME/.RLibrary
+                    mkdir -p $HOME/go
+                    curl -o- https://raw.githubusercontent.com\
+                    /creationix/nvm/v0.33.8/install.sh | bash
+                    echo 'export NVM_DIR=$HOME/.nvm' >> $BASH_ENV
+                    export NVM_DIR=$HOME/.nvm
+                    [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+                    nvm install 6.10.2
+                    nvm alias default node
+                    echo 'export PATH=/usr/lib/go-1.9/bin:$PATH' >> $BASH_ENV
```

Comment:
create a `pre:` step (similar to old circle) , which only does env and other *very* cheap commands.  installs should be in a separate step.

Ideally, put any *circle specific* preparation in a script `.circleci/image_setup.sh`.  i.e. commands which are related to the circle image `circleci/jdk8:0.1.1`.